Add mixin counts

cuyaproductions opened the pull request:
**Closes #57:** Add an additional option in the report mixin to output counts of total modules, test, and assertions found.

I know the issue was asking for number of assertions, but I figured that expanding out to modules and tests would be just as useful and more robust than just assertions. If you agree with this thought, I wonder if also counting and outputting `output` and `expect` usages would be useful as well.

I also updated the README docs to reflect the change.

mirisuzanne commented on sass/true/_results.scss:
```diff
@@ -31,6 +34,11 @@
     @include _true-message($message, 'debug');
   }
+
+  @if $stats {
```
Since `true-terminal-output` is a global setting phrased generically, it should probably have an impact on stats output as well? Or we should re-think the settings?

mirisuzanne replied:
Stats could also be added to the comments summary above.

cuyaproductions replied:
Okay I added the stats to the summary.

About the `$true-terminal-output` setting: There may be some context I am missing since the issue was last discussed months ago. In one of the comments you mentioned that getting these stats would be something that can be turn on or off. I don't know if the `$true-terminal-output` setting existed then, but now that we're discussing it I am not really seeing a point of having a `$stats` parameter to the mixin. I think just using the value `$true-terminal-output` would be fine (I think this is what you're suggesting). What do you think?

cuyaproductions commented on sass/true/_results.scss:
```diff
@@ -125,3 +133,50 @@ $_true-test-result: null;
+/// @access private
+@mixin _true-update-stats-count($type) {
+  @if index(map-keys($_true-stats-count), $type) {
+    $new-count: map-get($_true-stats-count, $type) + 1;
```
Thank you! That's a helpful function 😃

cuyaproductions commented on sass/true/_results.scss:
```diff
@@ -125,3 +133,50 @@ $_true-test-result: null;
+// -------------
+/// Stats message
+/// @access private
+/// @return {String} - Stats debug message
```
Ah yes sorry forgot.

cuyaproductions commented:
Looks like my comments are hidden in outdated code:

> Okay I added the stats to the summary.
> About the $true-terminal-output setting: There may be some context I am missing since the issue was last discussed months ago. In one of the comments you mentioned that getting these stats would be something that can be turn on or off. I don't know if the $true-terminal-output setting existed then, but now that we're discussing it I am not really seeing a point of having a $stats parameter to the mixin. I think just using the value $true-terminal-output would be fine (I think this is what you're suggesting). I updated this mixin to reflect what I think makes sense. Let me know if you agree.

Lastly, I also made updates to the true-cli code to handle the stats data and print along with summary data.

Let me know what you think!

mirisuzanne commented:
Sorry I've been busy. I'll take a look at it this week. Thanks!

mirisuzanne commented:
@cuyaproductions Sorry it took so long. I released 2.2.2 with your changes.

cuyaproductions commented:
🎉Yay!! No worries, @mirisuzanne. I'm happy I could contribute.
